### PR TITLE
DOC: correct linestyle example and reference rcParams

### PR DESCRIPTION
**Pull request by rcomer**
## PR summary

Today I had reason to want to know exactly what the "dotted" pattern is and thought this example could have been more helpful.  So I

- added a sentence about configuration by `rcParams`
- removed the statement that "dotted" is the same as the pattern (1, 1), since the default rcParam for that is (1, 1.65)
- in the tuple examples, increased the spacing for "dotted" as it was identical to "densely dotted".  I picked 5 to be consistent with the other examples that are not "densely" or "loosely".

## PR checklist

- [n/a] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [n/a] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [n/a] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [n/a] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by rcomer** [comment]
[Renders like this](https://output.circle-artifacts.com/output/job/52ff58a9-fc83-40da-b8d2-0b7510bb91ea/artifacts/0/doc/build/html/gallery/lines_bars_and_markers/linestyles.html#sphx-glr-gallery-lines-bars-and-markers-linestyles-py)

File: galleries/examples/lines_bars_and_markers/linestyles.py [hunk 1]
```diff
@@ -8,7 +8,9 @@
 ``(offset, (on_off_seq))``. For example, ``(0, (3, 10, 1, 15))`` means
 (3pt line, 10pt space, 1pt line, 15pt space) with no offset, while
 ``(5, (10, 3))``, means (10pt line, 3pt space), but skip the first 5pt line.
-See also `.Line2D.set_linestyle`.
+See also `.Line2D.set_linestyle`.  The specific on off sequence of the
```
Comment (jklymak) [inline galleries/examples/lines_bars_and_markers/linestyles.py:11]:
```suggestion
See also `.Line2D.set_linestyle`.  The specific on/off sequence of the
```

File: galleries/examples/lines_bars_and_markers/linestyles.py [hunk 2]
```diff
@@ -8,7 +8,9 @@
...
+"dotted", "dashed" and "dashdot" styles may be configured using
+``rcParams["lines.{style}_pattern"]``.
```
Comment (jklymak) [inline galleries/examples/lines_bars_and_markers/linestyles.py:13]: The f-string replacer here is a little mysterious - maybe add an example?
```suggestion
``rcParams["lines.{style}_pattern"]``, eg. `rcParams["lines.dotted_pattern"] = (0, (1, 10))`. 
```

Comment (rcomer) [reply galleries/examples/lines_bars_and_markers/linestyles.py]: What do you think about listing them all out with the `:rc:` directive?  I quite like that we can see the defaults for free, but not sure if it's just too much...

Comment (jklymak) [reply galleries/examples/lines_bars_and_markers/linestyles.py]: sure, explicit is probably fine.

**Comment by rcomer** [comment]
[Current iteration](https://output.circle-artifacts.com/output/job/5ec9f0a6-88e9-4bab-bfec-abb51c72d19c/artifacts/0/doc/build/html/gallery/lines_bars_and_markers/linestyles.html#sphx-glr-gallery-lines-bars-and-markers-linestyles-py)

**Comment by timhoffm** [comment]
The names and original pattern values are taken over from TikZ in #7087. That somehow got lost in #12586. But I suggest that we keep [these relations](https://tex.stackexchange.com/questions/45275/tikz-get-values-for-predefined-dash-patterns) and possibly also mention that again.

**Comment by QuLogic** [comment]
While you're here, can you shrink the figure width to under 7 inches? It is currently being scaled down to fit the page content width, which doesn't look good, and I don't see any reason it needs to be this wide specifically.

**Comment by timhoffm** [comment]
> I don't see any reason it needs to be this wide specifically.

Pydata-sphinx-theme has a smaller main content width than our previous website layout. There may well be a number of figures that don't fit anymore.

**Comment by QuLogic** [comment]
So I double checked on a wide image and the width is actually 896px, which at 100 dpi is just under 9 inches, so it shouldn't be too much different with constrained layout on.

I also went through our sphinx gallery results, and we have 76 images over the limit.

**Comment by jklymak** [comment]
I think sometimes it's fine to shrink an image.  Usually instead of shrinking all the fonts and linewidths.  In this case I'm sure the figure could simply be smaller.

**Comment by rcomer** [comment]
@timhoffm AFAICS the patterns we currently have here still match #7087, but not what is given in that StackExchange answer.  I note #7087 is newer than the SE answer, so maybe TikZ changed?

**Comment by rcomer** [comment]
[7 inch version](https://output.circle-artifacts.com/output/job/05729400-f8ac-45aa-bccc-3641150c2d18/artifacts/0/doc/build/html/gallery/lines_bars_and_markers/linestyles.html#sphx-glr-gallery-lines-bars-and-markers-linestyles-py)

**Comment by timhoffm** [comment]
I find it somewhat confusing that the names under  "named line styles" are repeated in the "parametrized linestyles " but have different patterns.Obviously, we cannot change the patterns for the matplotlib names. If they are different to the tikz patterns, we should clearly communicate that the above are Matplotlib named linestyles and that the pattern section are examples to reproduce the tikz linestyles.

**Comment by rcomer** [comment]
As I understand it, we specify our dash lengths in multiples of linewidth whereas Tikz specifies dash length in pixels (with only "dot" specified as linewidth).  So we can only go so far to match the Tikz styles.  We could put the Tikz numbers in and say these are "adapted from Tikz styles"?

**Comment by rcomer** [comment]
If the repeated name is the issue, we could also just update the parametrized "dotted" to e.g. "medium dotted".

**Review by timhoffm — APPROVED** [review]
Ok, let's not overthink this. We can still improve later.

**Pull request merged**